src/WasmParser.ts

```diff
@@ -808,6 +821,15 @@ export class BinaryReader {
       this.state = BinaryReaderState.END_WASM;
       return true;
     }
+    if (this._pos < this._length - 4) {
```

Comment:
more complex check is needed when _eof == false, `if (this._pos < this._length - (this._eof ? 4 : 1)) {`, then `if (magicNumber === (WASM_MAGIC_NUMBER& ((1 << (Math.min(4, this._length - this._pos) * 8)) - 1)))`